spynnaker7/pyNN/models/pynn_projection.py

```diff
@@ -27,7 +32,11 @@ def __init__(
             synapse_dynamics=None, rng=None):
         # pylint: disable=too-many-arguments, protected-access
         synapse_dynamics_stdp = None
-        if synapse_dynamics is None:
+        if synapse_dynamics is None or \
+                (isinstance(synapse_dynamics,
+                 AbstractSynapseDynamicsStructural) and
+                 isinstance(synapse_dynamics.super,
+                 AbstractStaticSynapseDynamics)):
             synapse_dynamics_stdp = SynapseDynamicsStatic()
```

Comment:
I dont understand why you want a static dynamics if youve got a structural dynamics. 

is this some sort of silent error handling? at which point, if it is, just raise an exception saying 0.7 dont supprot structual plastics.

Reply:
now ive read this a bit more, that super obscures the logic here (thus another reason why it needs changing) but this def reads as : if i dont have a dynamics, set me to static. If i have dynamics, and they are structural, and those faked inhirtance is static, then make it static. 

whereas if you fix your classes to not fake its inheritance, you make this readable by doing
if (synapse_dynamics is None or isinstance(synapse_dynamics, AbstractStaticSynapseDynamics)):	
    blah

much more readable, and less likely for future developers to not look at that and assume some sort of silent failure is occurring